### PR TITLE
[TOPIC-BLE-LLCP] Bluetooth: controller: Add Tx Queue

### PR DESCRIPTION
**Pull request by thoh-ot**
Tx Queue implementation and accompanying unit test.

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_tx_queue.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+void ull_tx_q_enqueue_ctrl(struct ull_tx_q *queue, struct node_tx *tx);
+
+/**
+ * @brief Dequeue a tx node from a tx queue.
+ *
+ * @param ull_tx_q Address of tx queue.
+ *
+ * @return Head tx node of the tx queue.
+ */
+struct node_tx *ull_tx_q_dequeue(struct ull_tx_q *queue);
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_tx_queue.h]: Please clarify that `ull_tx_q_dequeue` is ULL-only callable and memq construct will continue to be used between ULL and LLL?

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_tx_queue.h (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_tx_queue.h]: Agreed by @thot-ot.

**Review by cvinayak — APPROVED** [review]

**Comment by thoh-ot** [comment]
Just pushed a minor update adding the ull_tx_queue.c to the CMakeLists.txt

**Review by cvinayak — APPROVED** [review]

**Comment by thoh-ot** [comment]
Reopen to re-run ci

**Review by kruithofa — APPROVED** [review]

**Pull request merged**